### PR TITLE
Eliminate dead branches due to always-true or always-false conditions

### LINKED ISSUE #20599: Always true/false conditions on master
**Version Used**: 
Latest master

**Actual Behavior**:
The following conditions are always true/false.

On class [BinderFactory.BinderFactoryVisitor.cs](https://github.com/dotnet/roslyn/blob/master/src/Compilers/CSharp/Portable/Binder/BinderFactory.BinderFactoryVisitor.cs#L479), the highlighted `if` condition is always false as it duplicates the previous condition (which returns).

On class [AbstractTriviaFormatter.cs](https://github.com/dotnet/roslyn/blob/master/src/Workspaces/Core/Portable/Formatting/TriviaEngine/AbstractTriviaFormatter.cs#L731), `_firstLineBlank` is always false because when it is true the method returns on line 718.

On class [GraphBuilder.cs](https://github.com/dotnet/roslyn/blob/master/src/VisualStudio/Core/Def/Implementation/Progression/GraphBuilder.cs#L342), `typeSymbol` is always different from null because of the direct cast.

On class [SuggestedActionSetComparer.cs](https://github.com/dotnet/roslyn/blob/master/src/EditorFeatures/Core/Implementation/Suggestions/SuggestedActionSetComparer.cs#L21), the last bit of the `if` condition is always false because it is redundant with the previous check (== null is equivalent to !HasValue).

On class [CSharpRenameRewriterLanguageService.cs](https://github.com/dotnet/roslyn/blob/master/src/Workspaces/CSharp/Portable/Rename/CSharpRenameRewriterLanguageService.cs#L160), the variable `shouldComplexifyNode` is always true because the code is contained in a conditional block.

On class [ParameterHelpers.cs](https://github.com/dotnet/roslyn/blob/master/src/Compilers/CSharp/Portable/Symbols/Source/ParameterHelpers.cs#L215), the last or clause of the `else if` condition is always false because it was tested to true on the previous `else if` condition.

On class [SolutionState.cs](https://github.com/dotnet/roslyn/blob/master/src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.cs#L1131), the parameter `folders` is always different from `null` because there was an explicit check to null just before.

Cheers

### PR DESCRIPTION
**Pull request by sharwell**
Fixes #20599

### CODE REVIEW

File: src/VisualStudio/Core/Def/Implementation/Progression/GraphBuilder.cs [hunk 1]
```diff
@@ -388,8 +388,8 @@ private static void UpdateLabelsForNode(ISymbol symbol, Solution solution, Graph
                 case SymbolKind.Property:
                     node.Label = symbol.MetadataName;
 
-                    var propertySymbol = symbol as IPropertySymbol;
-                    if (propertySymbol != null && propertySymbol.IsIndexer && LanguageNames.CSharp == propertySymbol.Language)
+                    var propertySymbol = (IPropertySymbol)symbol;
+                    if (propertySymbol.IsIndexer && LanguageNames.CSharp == propertySymbol.Language)
```
Comment (sharwell) [inline src/VisualStudio/Core/Def/Implementation/Progression/GraphBuilder.cs]: :memo: This one wasn't pointed out, but I believe is valid and now matches the pattern for the rest of this switch statement.

**Comment by sharwell** [comment]
@dotnet/roslyn-ide and @dotnet/roslyn-compiler for review

**Comment by cston** [comment]
LGTM

File: src/EditorFeatures/Core/Implementation/Suggestions/SuggestedActionSetComparer.cs [hunk 2]
```diff
@@ -18,7 +18,7 @@ public SuggestedActionSetComparer(SnapshotPoint? targetPoint)
         private static int Distance(Span? textSpan, SnapshotPoint? targetPoint)
         {
             // If we don't have a text span or target point we cannot calculate the distance between them
-            if (textSpan == null || !textSpan.HasValue || targetPoint == null || !targetPoint.HasValue)
+            if (!textSpan.HasValue || !targetPoint.HasValue)
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/Implementation/Suggestions/SuggestedActionSetComparer.cs]: i'd personally prefer we keep the null checks, and remove the .HasValue checks here.

Comment (sharwell) [inline src/EditorFeatures/Core/Implementation/Suggestions/SuggestedActionSetComparer.cs]: Me too, and it's what I did first. But the prevailing style of the file is the opposite and that takes precedence outside of a style-changing PR.

**Pull request merged**